File: algo_prime_numbers/prime.py
#!/bin/env python

import sys
import math
import time

primes = []
limit = int(sys.argv[1])
quantity = 0
number = 1

def is_prime(number, primes, quantity):
    if not number % 2 and number != 2: return False
    sqrt = number**0.5
    if sqrt == math.floor(sqrt): return False
    divisor = prime_pos = 0
    while divisor < sqrt:
        prime_pos += 1
        if quantity <= prime_pos:
            break
        divisor = primes[prime_pos]
        if not number % divisor: return False
    return True

print ""
start_time = time.time()

while True:
    number += 1
    if is_prime(number, primes, quantity):
        primes.append(number)
        quantity += 1
        if quantity >= limit:
            break

print "\nElapsed time:", time.time() - start_time, "milliseconds"
print "Last prime number found:",primes[limit - 1];

File: algo_prime_numbers/prime.c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <sys/timeb.h>
#include <math.h>

int main(int argc, char *argv[]){
    long count = 0, number = 1, primePos, diff;
    int isNotPrime = 0;
    double s;

    uint32_t *primes, limit, divisor;
    struct timeb start, end;

    printf("Enter the number of prime numbers required: ");
    scanf("%d",&limit);
    primes = malloc(limit * sizeof *primes);

    ftime(&start);

    while (number++){
        if (number % 2) {
            if ((s = sqrt(number)) == floor(s)) continue;
            primePos = isNotPrime = 0;
            while ((divisor = primes[++primePos]) && divisor < s)
                if (isNotPrime |= !(number % divisor)) break;
            if (isNotPrime) continue;
        } else if (number != 2) continue;
        primes[count++] = number;
        if (count >= limit) break;
    }
    
    ftime(&end);

    diff = (long) (1000.0 * (end.time - start.time)
        + (end.millitm - start.millitm));

    printf("\nElapsed time: %u milliseconds\n", diff);
    printf("Last prime number found: %d\n", primes[limit - 1]);

    free(primes);

    //system("PAUSE");
    return 0;
}
